src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -80,6 +83,34 @@ class WorkspaceService(protected val argUserToken: WithAccessToken,
     RequestComplete(WorkspaceStorageCostEstimate(f"$$$estimate%.2f", bucketUsage.lastUpdated))
   }
 
+  def getStorageCostEstimateV2(workspaceNamespace: String,
+                               workspaceName: String
+  ): Future[WorkspaceStorageUsageAndCostEstimate] = {
+    for {
+      bucketUsage <- rawlsDAO.getBucketUsageV2(workspaceNamespace, workspaceName)
+    } yield {
+      // Convert bytes to GB since rate is based on GB.
+      val (totalBytes, totalEstimate) = bucketUsage.metrics.foldLeft((BigDecimal(0), BigDecimal(0))) {
+        case ((sumBytes, sumEstimate), metric) =>
+          val bytes = BigDecimal(metric.valueInBytes)
+          val estimate = bytes / (1024 * 1024 * 1024) * storagePriceList(metric.storageClass)
+          (sumBytes + metric.valueInBytes, sumEstimate + estimate)
+      }
+      WorkspaceStorageUsageAndCostEstimate(f"$$$totalEstimate%.2f", totalBytes, Some(DateTime.now()))
```

Comment:
I was working on catching NoSuchElementException in the case of unexpected storage classes, which would show a "key not found" error since it's not in the `storagePriceList` map, and reformatting it to a different error message that explains the problem.  But as I struggled to do proper error handling in scala, I began to wonder if we really need to watch for this particular error, or if it should just be thrown as is since it's not clear the user will care?